pkg/cloud/aws/actuators/machine/actuator.go

```diff
@@ -271,43 +271,16 @@ func getSubnetIDs(subnet providerconfigv1.AWSResourceReference, availabilityZone
 	return subnetIDs, nil
 }
 
-// CreateMachine starts a new AWS instance as described by the cluster and machine resources
-func (a *Actuator) CreateMachine(cluster *clusterv1.Cluster, machine *clusterv1.Machine) (*ec2.Instance, error) {
-	machineProviderConfig, err := ProviderConfigFromMachine(machine)
-	if err != nil {
-		glog.Errorf("error decoding MachineProviderConfig: %v", err)
-		return nil, err
-	}
-
-	credentialsSecretName := ""
-	if machineProviderConfig.CredentialsSecret != nil {
-		credentialsSecretName = machineProviderConfig.CredentialsSecret.Name
-	}
-	client, err := a.awsClientBuilder(a.kubeClient, credentialsSecretName, machine.Namespace, machineProviderConfig.Placement.Region)
-	if err != nil {
-		glog.Errorf("unable to obtain AWS client: %v", err)
-		return nil, fmt.Errorf("unable to obtain AWS client: %v", err)
-	}
-
-	// We explicitly do NOT want to remove stopped masters.
-	if !IsMaster(machine) {
-		// Prevent having a lot of stopped nodes sitting around.
-		err = a.removeStoppedMachine(machine, client)
-		if err != nil {
-			glog.Errorf("unable to remove stopped machines: %v", err)
-			return nil, fmt.Errorf("unable to remove stopped nodes: %v", err)
-		}
-	}
-
-	// Get AMI to use
-	var amiID *string
-	if machineProviderConfig.AMI.ID != nil {
-		amiID = machineProviderConfig.AMI.ID
+func getAMI(AMI providerconfigv1.AWSResourceReference, client awsclient.Client) (*string, error) {
+	if AMI.ID != nil {
```

Comment:
Can we put a unit test for this

Reply:
I plan to add unit tests for more functions/methods. It's more suitable for another PR.